tests/test_requests.py

```diff
@@ -482,15 +483,30 @@ def set_responses(self, responses):
 
     def make_response(self, content, status='200 OK', headers={}):
         """Make a fake http.client.HTTPResponse based on a byte stream."""
+        if not headers:
+            headers["Content-Type"] = "text/html"
+
         formatted_headers = '\r\n'.join(
             '{}: {}'.format(k, v) for k, v in headers.items()
         )
-        stream = 'HTTP/1.1 {}\r\n{}\r\n{}'.format(
+        stream = 'HTTP/1.1 {}\r\n{}\r\n\r\n{}'.format(
             status, formatted_headers, content,
         )
         sock = FakeSocket(stream.encode('utf8'))
-        response = http.client.HTTPResponse(sock)
-        response.begin()  # parse the stream
+        http_response = http.client.HTTPResponse(sock)
+        http_response.begin()
+
+        # Python versions below 3.7 expect a non-urllib3 response.
+        if sys.version_info.minor < 7:
```

Comment:
This is going to break oddly when Python 4 happens. One should never check the minor without also checking the major.

However, is this the right thing to check at all? Both requests and urllib3 are outside stdlib; it's not clear to my why the version of the Python interpreter is important. From a quick scan through Git history, is it possible that this is just the result of something like https://github.com/psf/requests/pull/6226 changing the return type, and 3.7 being the minimum supported Python for that version of requests, meaning tox pulls in an older version on 3.5 and 3.6?

Reply:
Right yeah obv should check maj version. I thought I had, must've just got lazy.

I think you're correct about requests changing the return type and v3.7 being min supported for the new type. Should it be testing the requests version instead?

Reply:
If it really is that PR, then yeah, I don't see much choice other than checking the requests version.

Reply:
Hmm looks like the real culprit is the change from `urrlib3 1.x` to `2.x`. Will dig into it.